From c286002808fb3f1665a739053c94c9697be42f0a Mon Sep 17 00:00:00 2001
From: MaJun <majun258@linux.alibaba.com>
Date: Sat, 8 Feb 2020 22:49:28 +0800
Subject: [PATCH] csky: Fix the compile error when use abiv1

Fix the bug of csky_regs, because this struct
has no member r1, but a1.

Signed-off-by: MaJun <majun258@linux.alibaba.com>
---
 linux/csky/get_scno.c | 2 +-
 linux/csky/set_scno.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/linux/csky/get_scno.c b/linux/csky/get_scno.c
index 9efb202..9d4397a 100644
--- a/linux/csky/get_scno.c
+++ b/linux/csky/get_scno.c
@@ -12,7 +12,7 @@ arch_get_scno(struct tcb *tcp)
 #if defined(__CSKYABIV2__)
 	tcp->scno = csky_regs.regs[3];
 #else
-	tcp->scno = csky_regs.r1;
+	tcp->scno = csky_regs.a1;
 #endif
 	return 1;
 }
diff --git a/linux/csky/set_scno.c b/linux/csky/set_scno.c
index 655d108..c1e1c18 100644
--- a/linux/csky/set_scno.c
+++ b/linux/csky/set_scno.c
@@ -13,7 +13,7 @@ arch_set_scno(struct tcb *tcp, kernel_ulong_t scno)
 #if defined(__CSKYABIV2__)
 	csky_regs.regs[3] = scno;
 #else
-	csky_regs.r1 = scno;
+	csky_regs.a1 = scno;
 #endif
 	return set_regs(tcp->pid);
 }
-- 
1.8.3.1

